src/modules/users/__tests__/users.controller.spec.ts

```diff
@@ -52,161 +53,313 @@ describe('modules/users/UsersController', () => {
+      const data = new UserDto({ name: 'Crysfel Villa', avatar: 'test.jpg', roles: ['Admin', 'SomethingElse'] });
+
+      usersService.findByAuth0Id = jest.fn(() => Promise.resolve(<User>{ _id: new ObjectIdMock('123'), roles: ['Member'] }));
+      usersService.findById = jest.fn(() => Promise.resolve(original));
+      usersService.update = jest.fn(() => Promise.resolve({ ok: 1 }));
+
+      expect(await usersController.update(request, params, data)).toEqual(response);
+      expect(usersService.update).toHaveBeenCalled();
+      expect(usersService.update).toHaveBeenCalledWith({
+        ...original,
+        avatar: 'test.jpg',
+        name: 'Crysfel Villa',
+      });
+    });
+
+    it('should update roles if is an admin', async () => {
+      const request = { user: { auth0Id: '123' } };
+      const response = { success: true }
+      const params = { id: 123 };
+      const original = <User>{ _id: new ObjectIdMock('123'), roles: ['Member'] }
+      const data = new UserDto({ name: 'Crysfel Villa', avatar: 'test.jpg', roles: ['Member', 'Admin'] });
+
+      usersService.findByAuth0Id = jest.fn(() => Promise.resolve(<User>{ _id: new ObjectIdMock('123'), roles: ['Admin'] }));
+      usersService.findById = jest.fn(() => Promise.resolve(original));
+      usersService.update = jest.fn(() => Promise.resolve({ ok: 1 }));
+
+      expect(await usersController.update(request, params, data)).toEqual(response);
+      expect(usersService.update).toHaveBeenCalled();
+      expect(usersService.update).toHaveBeenCalledWith({
+        ...original,
+        avatar: 'test.jpg',
+        name: 'Crysfel Villa',
+        roles: ['Member', 'Admin'],
+      });
+    });
+
+    it('should not update the email', async () => {
+      const request = { user: { auth0Id: '123' } };
+      const response = { success: true }
+      const params = { id: 123 };
+      const original = <User>{ _id: new ObjectIdMock('123'), email: 'test@test.com', roles: ['Member'] }
+      const data = new UserDto({ name: 'Crysfel Villa', email: 'should@ignore.com', avatar: 'test.jpg', roles: ['Member', 'Admin'] });
+
+      usersService.findByAuth0Id = jest.fn(() => Promise.resolve(<User>{ _id: new ObjectIdMock('123'), roles: ['Admin'] }));
+      usersService.findById = jest.fn(() => Promise.resolve(original));
+      usersService.update = jest.fn(() => Promise.resolve({ ok: 1 }));
+
+      expect(await usersController.update(request, params, data)).toEqual(response);
+      expect(usersService.update).toHaveBeenCalled();
+      expect(usersService.update).toHaveBeenCalledWith({
+        ...original,
+        avatar: 'test.jpg',
+        name: 'Crysfel Villa',
+        roles: ['Member', 'Admin'],
+      });
+    });
+  })
+
+  describe('remove', () => {
+    it('should return success when removing a valid user', async () => {
+      const request = {
+        user: { auth0Id: '1234' },
+      };
+      const params = { id: '1234' };
+      const response = { success: true };
```

Comment:
How do you feel about abstracting this out of each test and placing them up at the `describe` level instead? There's a lot of redundant code surrounding these three mocks.

Reply:
done!